adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.features
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+
+/**
+ * RDD holding run length encoded coverage in a FeatureRDD. CoverageRDD stores
+ * coverage counts at a ReferenceRegion in the score attribute of
+ * a feature.
+ *
+ * @param featureRDD FeatureRDD holding records for coverage and corresponding
+ * referenceRegion
+ */
+case class CoverageRDD(featureRDD: FeatureRDD) {
+
+  /**
+   * Java friendly save function. Automatically detects the output format.
+   *
+   * If the filename ends in ".bed", we write a BED file. If the file name ends
+   * in ".narrow[pP]eak", we save in the NarrowPeak format. If the file name
+   * ends in ".interval_list", we save in the interval list format. Else, we
+   * save as Parquet. These files are written as sharded text files.
+   *
+   * @param filePath The location to write the output.
+   */
+  def save(filePath: java.lang.String) = {
+    featureRDD.save(filePath)
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size.
+   *
+   * @param region ReferenceRegion to fetch overlapping coverage from
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def getCoverage(region: ReferenceRegion, bpPerBin: Int = 1): RDD[Coverage] = {
+    val coverage = featureRDD.filterByOverlappingRegion(region)
+
+    val flattenedCoverage = coverage.rdd.flatMap(r => {
+      val positions: List[Long] = List.range(r.getStart, r.getEnd)
+      positions.map(n => Coverage(r.getContigName, n, r.getScore))
+    }).filter(r => r.position >= region.start && r.position < region.end) // filter out positions from flanking regions
+
+    flattenedCoverage.filter(r => r.position % bpPerBin == 0)
+  }
+
+  /**
+   * Gets raw RDD of coverage
+   * @return RDD[Coverage] from underlying FeatureRDD
+   */
+  def rdd: RDD[Coverage] = {
+    featureRDD.rdd.flatMap(r => {
+      val positions: List[Long] = List.range(r.getStart, r.getEnd)
+      positions.map(n => Coverage(r.getContigName, n, r.getScore))
+    })
+  }
+}
+
+/**
+ * Coverage Recordfor CoverageRDD
+ * @param referenceName Specifies chromosomal location of coverage
+ * @param position Specifies position of coverage
+ * @param count Specifies count of coverage at location
+ */
+case class Coverage(referenceName: String, position: Long, count: Double)
```

Comment:
There is already a Coverage class. See [here](https://github.com/bigdatagenomics/adam/blob/1d0ace90f5f943704908550136e69c44ba357450/adam-core/src/main/scala/org/bdgenomics/adam/rdd/Coverage.scala). We do not seem to use this/ these are very different. Is there anything difference I can name my Coverage?